prow/tide/tide.go

```diff
@@ -188,7 +188,12 @@ func (c *Controller) Sync() error {
 		if err != nil {
 			return err
 		}
-		pool = append(pool, poolPRs...)
+		for _, pr := range poolPRs {
+			// Only keep PRs that are mergeable or haven't had mergeability computed.
```

Comment:
Won't this cause the same bug if the state hasn't been computed but once it is, it will be conflicting?

Reply:
Supposedly its occurence will be less lower now.

Reply:
But why not just accept only mergeable PRs?

Reply:
I don't want to ignore PRs just because they haven't had their mergeability calculated. That could cause PRs to temporarily disappear from the pool whenever their mergeability needs to be recalculated which would be confusing behavior for users.
If a PR has unknown mergeability, but is actually unmergeable we will likely find out before we actually try to merge it because subsequent sync loops should have `mergeable` set to false (the background job on GH will have had time to complete).
In the worst case scenario, a PR's mergeability becomes unknown the same sync loop when we want to merge it. In this case we would fail to merge the PR, but the next sync loop the `mergeable` field should be set to false. Previously we continued to consider the PR, but now it should be kicked out of the pool during the next sync loop because its mergeability will be known to be false.

Reply:
ACK